executor.go

```diff
@@ -389,7 +389,27 @@ func (e *executor) runJob(j internalJob, jIn jobIn) {
 		}
 		defer func() { _ = lock.Unlock(j.ctx) }()
 	}
-	_ = callJobFuncWithParams(j.beforeJobRuns, j.id, j.name)
+
+	// execute before func, unless user has specified BeforeJobRunsWithError
+	var beforeFunc any
+
+	beforeFunc = j.beforeJobRuns
+
+	if j.beforeJobRunsWithError != nil {
+		beforeFunc = j.beforeJobRunsWithError
+	}
```

Comment:
I don't see a reason that this needs to override the existing `beforeJobRuns`. I see the following scenarios as all valid:
```
1. Not running either beforeJobRuns and beforeJobRunsSkipIfBeforeFuncErrors
2. Running either beforeJobRuns or beforeJobRunsSkipIfBeforeFuncErrors
3. Running both beforeJobRuns and beforeJobRunsSkipIfBeforeFuncErrors
```

I would have it run the `beforeJobRuns` func first. Note that the function `callJobFuncWithParams` doesn't do anything if the passed function is nil, which it will be if unset. Then run the `beforeJobRunsSkipIfBeforeFuncErrors` and do the logic you have below to reschedule and return if an error is returned.

Reply:
neat! even better - will incorporate